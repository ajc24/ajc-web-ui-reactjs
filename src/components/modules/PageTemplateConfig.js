/**
 * Developed by Anthony Cox in 2025
 */
import { storybookCssClassList } from '../data/storybook';

/* Commonly used CSS classes */
const bodyDefaultCss = 'body-default';

/**
 * Functionality designed to work with the page template components (headers, menu bars, body content, footers)
 */
export default class PageTemplateConfig {
  /**
   * Setup and configure the current web page to ensure the default CSS styling is set to it.
   * This also handles the styling of pages which are deployed on Storybook
   */
  static setupDocumentBodyCss() {
    /* If we are rendering on Storybook, remove the Storybook classes which will affect the width of the component */
    for (let index = 0; index < storybookCssClassList.length; index += 1) {
      if (document.body.classList.contains(storybookCssClassList[index]) === true) {
        document.body.classList.remove(storybookCssClassList[index]);
      }
    }
    /* Ensure that the <body> element stretches to 100% of the width of the screen, that the X axis scrollbar is hidden and Y axis scrollbar is always displayed */
    if (document.body.classList.contains(bodyDefaultCss) === false) {
      document.body.classList.add(bodyDefaultCss);
      document.activeElement.blur();
    }
  }
}
