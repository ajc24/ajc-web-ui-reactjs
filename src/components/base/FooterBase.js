/**
 * Developed by Anthony Cox in 2025
 */
import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import PageTemplateConfig from '../modules/PageTemplateConfig';
import '../css/common.css';
import './css/base-footer.css';

const defaultComponentId = 'default-id--footer-base';

/**
 * Baseline component setting the template for all Footer components to be developed from.
 * This baseline auto-handles screen widths from the most commonly used mobile screen sizes (360x800) to the most commonly used desktop sizes (1920x1080).
 * The minimum height of the component is 71px, expanding to whatever height is suitable for the content within.
 */
const FooterBase = props => {
  const [id, setId] = useState(defaultComponentId);

  useEffect(() => {
    /* Setup the default CSS styling for the page which is rendering this component */
    PageTemplateConfig.setupDocumentBodyCss();

    /* Set the ID for the component */
    if (props.id !== undefined) {
      setId(`${props.id}--footer-base`);
    }
  });

  /* Set the styling for the navigation element */
  const footerCss = 'footer screen-width-root background-white';

  /* Set the styling for the outer content element */
  let outerContentCss = 'footer-content-outer screen-width-content-outer';
  if (props.backgroundColour === 'white' || props.backgroundColour === 'red' || props.backgroundColour === 'green') {
    /* Set the background colour to the specified colour */
    outerContentCss += ` background-${props.backgroundColour}`;
  } else if (props.backgroundColour === 'grey') {
    /* Set the background colour as grey */
    outerContentCss += ` background-${props.backgroundColour}-2`;
  } else {
    /* Default set the background colour as white */
    outerContentCss += ` background-white`;
  }
  /* Set the styling for the inner content element */
  const innerContentCss = 'footer-content-inner screen-width-content-inner';
  return (
    <footer role="contentinfo" id={id} className={footerCss}>
      <div className={outerContentCss}>
        <div className={innerContentCss}>
          {props.children}
        </div>
      </div>
    </footer>
  );
}
FooterBase.propTypes = {
  /** The background colour for the footer. The default colour for the background is white. */
  backgroundColour: PropTypes.oneOf([ 'green', 'grey', 'red', 'white' ]),
  /** The content to be displayed within the footer template component. */
  children: PropTypes.any,
  /** The unique identifier for this component. */
  id: PropTypes.string,
};
export default FooterBase;
