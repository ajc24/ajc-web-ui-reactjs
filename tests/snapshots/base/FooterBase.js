/**
 * Developed by Anthony Cox in 2024
 */
import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-testing-code';
import FooterBase from '../../../src/components/base/FooterBase';
import PageTemplateConfig from '../../../src/components/modules/PageTemplateConfig';

describe('Footer Base', () => {
  /* Set the spies for use in the tests */
  let setupDocumentBodyCssSpy;

  beforeAll(() => {
    setupDocumentBodyCssSpy = jest
      .spyOn(PageTemplateConfig, 'setupDocumentBodyCss')
      .mockImplementation(() => {});
  });

  afterAll(() => {
    setupDocumentBodyCssSpy.mockRestore();
  });

  describe('Component with default id and default size', () => {
    let snapshot;
    
    beforeAll(() => {
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <FooterBase>
            Footer base text content.
          </FooterBase>
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Component with custom id and small size', () => {
    let snapshot;
    
    beforeAll(() => {
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <FooterBase id="custom-id" size="small">
            Footer base text content.
          </FooterBase>
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Component with default id and tall size', () => {
    let snapshot;
    
    beforeAll(() => {
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <FooterBase size="tall">
            Footer base text content.
          </FooterBase>
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });
});
