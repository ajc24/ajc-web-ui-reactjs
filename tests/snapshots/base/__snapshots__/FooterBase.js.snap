// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Base Component with custom id and small size verifies the snapshot for the component 1`] = `"<footer role="contentinfo" id="custom-id--footer-container" class="footer-container background-white footer-container-size-small"><div class="footer-content-container background-transparent">Footer base text content.</div></footer>"`;

exports[`Footer Base Component with default id and default size verifies the snapshot for the component 1`] = `"<footer role="contentinfo" id="default-id--footer-container" class="footer-container background-white footer-container-size-small"><div class="footer-content-container background-transparent">Footer base text content.</div></footer>"`;

exports[`Footer Base Component with default id and tall size verifies the snapshot for the component 1`] = `"<footer role="contentinfo" id="default-id--footer-container" class="footer-container background-white footer-container-size-tall"><div class="footer-content-container background-transparent">Footer base text content.</div></footer>"`;
